src/AgentCommerce/Validation/ValidationIssues.php

```diff
@@ -0,0 +1,68 @@
+<?php declare(strict_types=1);
+/*
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Swag\PayPal\AgentCommerce\Validation;
+
+use Shopware\Core\Checkout\Cart\LineItem\LineItem;
+use Shopware\Core\Content\Product\ProductEntity;
+use Shopware\Core\Framework\Adapter\Translation\AbstractTranslator;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\PayPalSDK\Builder\AgenticCommerce\V1\ValidationIssueBuilder;
+use Shopware\PayPalSDK\Struct\AgenticCommerce\V1\Context\InventoryIssueContext;
+use Shopware\PayPalSDK\Struct\AgenticCommerce\V1\ResolutionOption;
+use Shopware\PayPalSDK\Struct\AgenticCommerce\V1\ValidationIssue;
+
+#[Package('checkout')]
+class ValidationIssues
+{
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly AbstractTranslator $translator,
+    ) {
+    }
+
+    public function outOfStock(LineItem $item, ?ProductEntity $restockProduct): ValidationIssue
+    {
+        $builder = new ValidationIssueBuilder();
+
+        $builder
+            ->withCode(ValidationIssue::CODE__INVENTORY_ISSUE)
+            ->withType(ValidationIssue::TYPE__BUSINESS_RULE)
+            ->withMessage($this->translator->trans('swag_paypal.agent_commerce.validation_issue.out_of_stock.message'))
+            ->withUserMessage($this->translator->trans('swag_paypal.agent_commerce.validation_issue.out_of_stock.user_message'))
+            ->withItemId($item->getPayloadValue('productNumber')) // @phpstan-ignore method.deprecated
+            ->addResolutionOption()
+            ->withAction(ResolutionOption::ACTION__REMOVE_ITEM)
+            ->withLabel('swag_paypal.agent_commerce.validation_issue.out_of_stock.resolution_option.remove.label')
+            ->withMetadata()
+            ->withCostImpact((string) (-1 * $item->getPrice()?->getTotalPrice()))
+            ->end()
```

Comment:
```suggestion
                ->withAction(ResolutionOption::ACTION__REMOVE_ITEM)
                ->withLabel('swag_paypal.agent_commerce.validation_issue.out_of_stock.resolution_option.remove.label')
                ->withMetadata()
                ->withCostImpact((string) (-1 * $item->getPrice()?->getTotalPrice()))
                ->end()
```

Reply:
Can we check that? The whole purpose of the builders is to add readability for these complex types